python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py

```diff
@@ -418,25 +418,23 @@ def _get_step_records() -> Sequence[EventLogEntry]:
             raise
 
     def _grant_permissions(
-        self, log: DagsterLogManager, databricks_run_id: int, request_retries: int = 3
+        self, log: DagsterLogManager, databricks_run_id: int, request_retries: int = 6
     ) -> None:
         client = self.databricks_runner.client.workspace_client
         # Retrieve run info
         cluster_id = None
         for i in range(1, request_retries + 1):
             run_info = client.jobs.get_run(databricks_run_id)
-            if run_info.cluster_instance is None:
-                check.failed("Databricks run {databricks_run_id} has null cluster_instance")
             # if a new job cluster is created, the cluster_instance key may not be immediately present in the run response
             try:
-                cluster_id = run_info.cluster_instance.cluster_id
+                cluster_id = run_info.tasks[0].cluster_instance.cluster_id
```

Comment:
since Databricks Jobs v2.1 API, the `cluster_instance` is set on the individual task of a job run. The `run_info.cluster_instance` is always returned as None. I believe it exists for backward compatibility. Since the dagster_databricks client is using the new v2.1 format, we need to access `cluster_info` for the run from the `tasks` object.

![Screenshot 2024-10-04 at 10 41 35 AM](https://github.com/user-attachments/assets/86418109-4a0e-4ec7-ba58-dcaf5c276ce3)
![Screenshot 2024-10-04 at 10 40 41 AM](https://github.com/user-attachments/assets/696ef577-da41-48fb-ba2b-f6e292bfafc0)